### PR TITLE
DAOS-13392 test: Add fault injection metrics into Jenkins

### PR DESCRIPTION
**Pull request by ashleypittman**
Use the log analysis tools to record what logging daos
performed and save this to a XML file so we can report in
Jenkins what code was tested by NLT.

Required-githooks: true

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: src/tests/ftest/cart/util/cart_logtest.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env python3
 #
-# Copyright 2018-2023 Intel Corporation
+# (C) Copyright 2018-2023 Intel Corporation
```
Comment (JohnMalmberg) [inline src/tests/ftest/cart/util/cart_logtest.py:3]: Any specific reason for adding the "(C)"?  It has no legal value in copyright law.

**Review by JohnMalmberg — APPROVED** [review]

File: src/tests/ftest/cart/util/cart_logtest.py (hunk 1 above)
Comment (ashleypittman) [reply src/tests/ftest/cart/util/cart_logtest.py:3]: The commit hook wasn't triggering for some reason so I changed it to be consistent with other files but I'm not sure if it actually helped.  Will revert this line on next push.

**Review by brianjmurrell — APPROVED** [review]

File: Jenkinsfile [hunk 2]
```diff
@@ -1057,7 +1061,11 @@ pipeline {
                             sconsBuild(parallel_build: true,
                                        scons_args: 'PREFIX=/opt/daos TARGET_TYPE=release BUILD_TYPE=debug',
                                        build_deps: 'no'))
+                        unstash('nltr')
```
Comment (brianjmurrell) [inline Jenkinsfile:1064]: Realizing that this has landed and I +1'd it, it's just becoming clear that the ramifications of this are that if one skips the UT stages, this stage fails as it cannot find the `nltr` stash.

Comment (ashleypittman) [reply Jenkinsfile:1064]: Yes, I was hoping to speak to you about that.  The code in the FI stage is written to work, and the NLT python code will print an error but not fail if the file isn't present.  The only thing that's missing is a try/catch around the unstash command and you'll see from previous versions of this PR I couldn't get that to work.

**Pull request merged**